### PR TITLE
client: use httputil.LoggedTransport in client as well

### PR DESCRIPTION
**Pull request by mvo5**

### CODE REVIEW

File: httputil/logger.go [hunk 1]
```diff
@@ -55,22 +55,22 @@ func (f debugflag) debugBody() bool {
 type LoggedTransport struct {
 	Transport http.RoundTripper
 	Key       string
-	body      bool
+	Body      bool
```
Comment (MiguelPires) [inline httputil/logger.go:58]: It wasn't clear what this was from the `Body: true` usage. Maybe `LogBody` or `MayLogBody`?

File: client/client.go [hunk 2]
```diff
@@ -120,8 +128,14 @@ func New(config *Config) *Client {
...
 	}
 	return &Client{
-		baseURL:     *baseURL,
-		doer:        &http.Client{Transport: &http.Transport{DisableKeepAlives: config.DisableKeepAlive}},
+		baseURL: *baseURL,
+		doer: &http.Client{Transport: &httputil.LoggedTransport{
+			Transport: &http.Transport{
+				DisableKeepAlives: config.DisableKeepAlive,
+			},
+			Key:  "SNAP_CLIENT_DEBUG_HTTP",
```
Comment (MiguelPires) [inline client/client.go:136]: This is missing a test checking that setting the env var enables the logs

File: client/client.go [hunk 3]
```diff
@@ -120,8 +128,14 @@ func New(config *Config) *Client {
...
+			Body: true,
```
Comment (pedronis) [inline client/client.go:137]: do we need to do somethin special about InstallPath ? I notice that the store package code sets up a client where this is false when doing downloads for example

**Review by pedronis — COMMENTED** [review]
a question about body logging vs InstallPath/snap uloading

**Comment by MiguelPires** [comment]
@mvo5 can I steal this PR from you? I wanted to use something like it today and remembered you opened this. So if you're busy, I can address the reviews and get it merged

**Comment by mvo5** [comment]
Closing as this is superseeded by https://github.com/snapcore/snapd/pull/12424

**Comment by MiguelPires** [comment]
> Closing as this is superseeded by #12424

Just so we have a correct trail, that PR was fixing the "we can't use SNAPD_DEBUG_HTTP" w/o setting SNAPD_DEBUG problem. I was waiting for it to be merged since otherwise the new env var would also have the same issue. I'll update this comment once I open the new version of this

**Pull request closed without merging**